### DISCUSSION TITLE
DOM4J and JFree cannot be resolved

### DISCUSSION THREAD

**Discussion by rnveach** [checkstyle/eclipse-cs#455] [Q&A]
When I import Eclipse-CS into Eclipse to compile from the source, I get the following errors (trimmed).

````
Description	Resource	Path	Location	Type
src/main/java/ is missing from source..	build.properties	/net.sf.eclipsecs.checkstyle	line 6	Plug-in Problem
No available bundle exports package 'org.dom4j'	MANIFEST.MF	/net.sf.eclipsecs.core/META-INF	line 25	Plug-in Problem
Unsatisfied constraint: 'Import-Package: org.yaml.snakeyaml; version="1.33.0"'	MANIFEST.MF	/net.sf.eclipsecs.core/META-INF	line 45	Plug-in Problem
The import org.dom4j cannot be resolved	XMLUtil.java	/net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/util	line 28	Java Problem
The import org.jfree cannot be resolved	GraphPieDataset.java	/net.sf.eclipsecs.ui/src/net/sf/eclipsecs/ui/stats/views	line 32	Java Problem
````

I am using Eclipse 2022-06 for Enterprise Java and Web Developers. I tried using "for RCP and RAP Developers" but it gave the same results.

I have no issues building the source from command line maven, but I would prefer to be able to work with the source in the IDE.

@Calixte Do you know what I need to do for the project to compile in Eclipse?

Edit:
I have no issues compiling 10.4 in eclipse. It looks like the errors are related to 32d604929ad7f24488cd175e2f587b49986d5056 .

**Comment by rnveach** [comment]
I think I resolved this messing around.

I had to open `net.sf.eclipsecs.target\net.sf.eclipsecs.target.target` and click on the top right `Set As Active Target Platform` on the `Definition` tab. Clean, recompile and the suite no longer has any builder errors.

@Calixte Let me know if this is not right.

Edit: Seems there may be a bug in eclipse, where you may have to clean osgi and possibly the metadata if it doesn't succeed the first time for some reason.

**Reply by Bananeweizen** [reply]
Please have a look at ...eclipsecs.target\readme.md. Feel free to edit if you feel something is missing.

**Reply by rnveach** [reply]
I will look it over. It wasn't linked from the main readme so I didn't really see it. I am seeing `target` is a common thing for eclipse plugins and am still just getting to know these things.
